src/pages/StocktakeEditPage.js

```diff
@@ -159,15 +158,18 @@ export class StocktakeEditPage extends React.Component {
    */
   onEndEditing = (key, stocktakeItem, newValue) => {
     const { database } = this.props;
-    const { reasons } = this.state;
+    const { reasons, isReasonsModalOpen } = this.state;
 
     if (key !== 'countedTotalQuantity' || newValue === '') return;
     const quantity = parsePositiveInteger(newValue);
     if (quantity === null) return;
+    // If the reason modal is open just ignore any change to the current line
+    // This a hack to solve https://github.com/openmsupply/mobile/issues/1011
```

Comment:
Don't think we should tie our codebase to our version control:

```
// This a hack to ensure the current stock take item is not unexpectedly
// updated by the user immediately before or while the reasons modal is open.
```

Reply:
Annoyingly you can't make a multiline suggestion in github